cmd/state-tray/internal/open/open_lin.go

```diff
@@ -2,12 +2,29 @@
 
 package open
 
-import "errors"
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/ActiveState/cli/internal/locale"
+)
 
 func Prompt(command string) error {
-	return errors.New("not implemented")
-}
+	shellData, err := exec.Command("bash", "-c",
+		"grep ^$(id -un): /etc/passwd | cut -d: -f7-",
+	).Output()
+	if err != nil {
+		return locale.WrapError(err,
+			"err_determine_default_shell", "Could not determine default shell",
+		)
+	}
+	shell := string(shellData[:len(shellData)-1]) // trim newline
+
+	command = fmt.Sprintf("%s;%s", command, shell)
+	cmd := exec.Command("x-terminal-emulator", "-e", shell, "-c", command)
```

Comment:
Why do we need to feed the shell into this at all? Shouldn't the terminal emulator pick that up?

Reply:
When the terminal is brought up without a shell, it will execute the command and collapse immediately. The user will not see the output unless it is held it open. `x-terminal-emulator --hold` does work to display the output, but then the process hangs until it's signaled in some way. This signal cannot be initiated by users using Ctrl-{C,X,Z,D} from within the opened terminal.

The code here is the first reliable way I found to nicely open a terminal with the shell that the user configured. I could dig more for an alternative if you'd like.